WayMovies/Model/Request.swift

```diff
@@ -0,0 +1,305 @@
+     `Parameter searchTerms: The search string given by the user`
+     */
+    static func searchPeople(searchTerms: String, completion: @escaping(Result<[TVItem], Error>) -> Void) {
+        let searchURL = URL(string: "https://api.themoviedb.org/3/search/person?api_key=\(API_KEY)&query=\(replaceSpaces(string: searchTerms))")!
+        
+        let task = URLSession.shared.dataTask(with: searchURL) { (data, response, error) in
+            guard let jsonData = data else {
+                completion(.failure(error!))
+                return
+            }
+            
+            do {
+                let decoder = JSONDecoder()
+                let searchResponse = try decoder.decode(Response.self, from: jsonData)
+                let tvItems = searchResponse.results
+                completion(.success(tvItems))
+                
+            } catch {
+                completion(.failure(error))
+            }
+            
+        }
+        task.resume()
+    }
+    
+    /**
+     Searches the list of shows on TMDB with the given terms and returns a relevant list of results as TVItems
+     `Parameter searchTerms: The search string given by the user`
+     */
+    static func searchShows(searchTerms: String, completion: @escaping(Result<[TVItem], Error>) -> Void) {
+        let searchURL = URL(string: "https://api.themoviedb.org/3/search/tv?api_key=\(API_KEY)&query=\(replaceSpaces(string: searchTerms))")!
+        
+        let task = URLSession.shared.dataTask(with: searchURL) { (data, response, error) in
+            guard let jsonData = data else {
+                completion(.failure(error!))
+                return
+            }
+            
+            do {
+                let decoder = JSONDecoder()
+                let searchResponse = try decoder.decode(Response.self, from: jsonData)
+                let tvItems = searchResponse.results
+                completion(.success(tvItems))
+                
+            } catch {
+                completion(.failure(error))
+            }
+            
+        }
+        task.resume()
+    }
+    
+    /**
+     Gets the image data from the given image URL
+     `Parameter searchTerms: The url location of the image`
+     */
+    static func getImage(searchTerms: String, completion: @escaping(Result<Data, Error>) -> Void) {
+        
+        let searchURL = URL(string: "https://image.tmdb.org/t/p/original\(searchTerms)")!
+
+        let task = URLSession.shared.dataTask(with: searchURL) { (data, response, error) in
+            guard data != nil else {
+                completion(.failure(error!))
+                return
+            }
+            completion(.success(data!))
+        }
+        task.resume()
+    }
+    
+    /**
+     Replaces spaces in the given string with %20 for use in API get requests
+     */
+    static func replaceSpaces(string:String) -> String {
+        return string.replacingOccurrences(of: " ", with: "%20")
```

Comment:
in swift 5 you can simple add `var urlString = originalString.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed)` to avoid having to manually replace characters with string encodings